### ISSUE TITLE
[WARN] serf: Intent queue depth exceeds limit, dropping messages!

### ISSUE THREAD

**Issue by pierresouchay** [hashicorp/consul#7960]
When filing a bug, please include the following headings if possible. Any example text in this template can be deleted.

#### Overview of the Issue

In one of our preprod DCs, we have this message very often on Consul servers:
```
[WARN] serf: Intent queue depth (4321) exceeds limit (4096), dropping messages!
```
(with  Intent queue depth slighly different)

This message is not supposed to be displayed in such recent versions of Consul according to https://support.hashicorp.com/hc/en-us/articles/115015603408-Consul-Errors-And-Warnings 

Any hint on how fixing this?

### Consul info for both Client and Server

1.5.1criteoC -> 2 members
1.6.5 -> 409 members
1.6.5+entcriteo1 -> 6 entreprise servers (5 voting, 1 non-voting)

### Operating system and Environment details

CentOS 7, x64
a few Windows nodes running 1.6.5

**Comment by dnephin** [comment]
The two relevant code blocks for that warning are:
* https://github.com/hashicorp/consul/blob/master/vendor/github.com/hashicorp/serf/serf/serf.go#L1621-L1657
* https://github.com/hashicorp/consul/blob/master/lib/serf.go#L14-L19

It sounds like the older warning was skipped by setting a high warning value, but there is still a dynamic limit that causes messages to be dropped.

The `MinQueueDepth` is that limit, which is set to 4096.

**Comment by t0klian** [comment]
We are observing the same behaviour of the Consul (in relation to #8179 )

**Comment by banks** [comment]
Hi,

We believe based on an investigation of a similar issue (not reported on GitHub) that we understand why this is happening and will be releasing a fix as soon as possible. More details to follow!

**Comment by rboyer** [comment]
This will be fixed and backported to consul 1.8.x, 1.7.x, and 1.6.x shortly. The details of the underlying bug can be found: https://github.com/hashicorp/serf/pull/606

**Comment by orarnon** [comment]
Hi,
I'm having the same problem with version 1.13.1
